include/envoy/server/overload_manager.h

```diff
@@ -46,6 +65,12 @@ class OverloadManager {
    */
   virtual void registerForAction(const std::string& action, Event::Dispatcher& dispatcher,
                                  OverloadActionCb callback) PURE;
+
+  /**
+   * Get the thread-local overload action state cache. Lookups in this cache can be used as
+   * an alternative to registering a callback for overload action state changes.
+   */
+  virtual const OverloadActionStateCache& getOverloadActionStateCache() PURE;
```

Comment:
Nit: I might name this `getOverLoadThreadLocalState()`; "cache" is an overloaded concept and pointing out this is the thread-local variant makes the code a bit easier to read `</bikeshed>`.

Reply:
Changed to ThreadLocalOverloadState